app.js

```diff
@@ -1,36 +1,125 @@
+  this.itemPath = itemPath;
+  this.itemClick = 0;
+  this.itemShownTotal = 0;
+  labelNameArray.push(itemName);
+  itemsListArray.push(this);
+  // this.itemClick = 0;.
+}
+var a = new Item ('bag', 'assets/bag.jpg');
+var b = new Item ('banana', 'assets/banana.jpg');
+var c = new Item ('bathroom', 'assets/bathroom.jpg');
+var d = new Item ('boots', 'assets/boots.jpg');
+var e = new Item ('breakfast', 'assets/breakfast.jpg');
+var f = new Item ('bubblegum', 'assets/bubblegum.jpg');
+var g = new Item ('chair', 'assets/chair.jpg');
+var h = new Item ('cthulhu', 'assets/cthulhu.jpg');
+var i = new Item ('dog-duck', 'assets/dog-duck.jpg');
+var j = new Item ('dragon', 'assets/dragon.jpg');
+var k = new Item ('pen', 'assets/pen.jpg');
+var l = new Item ('pet-sweep', 'assets/pet-sweep.jpg');
+var m = new Item ('scissors', 'assets/scissors.jpg');
+var n = new Item ('shark', 'assets/shark.jpg');
+var o = new Item ('sweep', 'assets/sweep.png');
+var p = new Item ('tauntaun', 'assets/tauntaun.jpg');
+var q = new Item ('unicorn', 'assets/unicorn.jpg');
+var r = new Item ('usb', 'assets/usb.gif');
+var s = new Item ('water-can', 'assets/water-can.jpg');
+var t = new Item ('wine-glass', 'assets/wine-glass.jpg');
+
+function randomItemSelectionFunc(){
+  return Math.floor(Math.random() * (itemsListArray.length));
+};
+function randomPictureGenerator(){
+  var currentlyShownUserPageArray = [];
+  while (currentlyShownUserPageArray.length < 3) {
+    var randomItemSelectionVar = randomItemSelectionFunc();
+    if (!previouslyShownUserPageArray.includes(randomItemSelectionVar) && !currentlyShownUserPageArray.includes(randomItemSelectionVar)){
+      currentlyShownUserPageArray.push(randomItemSelectionVar);
+    }
+  }
+  var imageLeft = itemsListArray[currentlyShownUserPageArray[0]];
+  var imageCenter = itemsListArray[currentlyShownUserPageArray[1]];
+  var imageRight = itemsListArray[currentlyShownUserPageArray[2]];
+  image1.src = imageLeft.itemPath;
+  image2.src = imageCenter.itemPath;
+  image3.src = imageRight.itemPath;
+
+  image1.alt = currentlyShownUserPageArray[0];
+  image2.alt = currentlyShownUserPageArray[1];
+  image3.alt = currentlyShownUserPageArray[2];
+
+  previouslyShownUserPageArray = currentlyShownUserPageArray;
+
+  imageLeft.itemShownTotal++;
+  imageCenter.itemShownTotal++;
+  imageRight.itemShownTotal++;
+};
+randomPictureGenerator();
+
+function handleTheClick(){
+  randomPictureGenerator();
+  totalClicks++;
+  var productIndex = this.alt; // get the index of the product clicked
+  itemsListArray[productIndex].itemClick++;
+
+  if (totalClicks === clickLimit){
+    image1.removeEventListener('click', handleTheClick);
+    image2.removeEventListener('click', handleTheClick);
+    image3.removeEventListener('click', handleTheClick);
+    productClicks();
+  }
+}
+
+image1.addEventListener('click', handleTheClick);
+image2.addEventListener('click', handleTheClick);
+image3.addEventListener('click', handleTheClick);

```

Comment:
Where are `image1`, `image2`, and `image3` defined in your code? Some browsers will automatically pick up IDs on objects as independent variables that can be accessed, but not every browser acts that way. Use `document.getElementById`